src/logic/safe/utils/getFeeHistory.ts

```diff
@@ -0,0 +1,39 @@
+import { FeeHistoryResult } from 'web3-eth'
+import { hexToNumber } from 'web3-utils'
+
+import { getWeb3ReadOnly } from 'src/logic/wallets/getWeb3'
+import { logError, Errors } from 'src/logic/exceptions/CodedException'
+
+const DEFAULT_MAX_PRIO_FEE = 2.5e9 // 2.5 GWEI
+
+export const clampMaxPrioFeePerGas = (maxPriorityFeePerGas: number, maxFeePerGas: number): number => {
+  if (maxPriorityFeePerGas > maxFeePerGas) {
+    const minMaxPrioFee = 0.001
+    const maxMaxPrioFee = maxFeePerGas - 1
+
+    return Math.max(minMaxPrioFee, maxMaxPrioFee)
+  }
+
+  return maxPriorityFeePerGas
+}
+
+export const getMaxPriorityFeePerGas = async (): Promise<number> => {
+  let blocks: FeeHistoryResult | undefined
+  let maxPriorityFeePerGas: number | undefined
+
+  const web3 = getWeb3ReadOnly()
+
+  try {
+    // Lastest block, 50th reward percentile
+    blocks = await web3.eth.getFeeHistory(1, 'latest', [50])
+    maxPriorityFeePerGas = hexToNumber(blocks.reward[0][0])
```

Comment:
Can this be moved to after line 34 where we check that `blocks` exists? We could simplify the condition in line 34 then and set this as a const

Reply:
This is not necessary after the recent changes I made.